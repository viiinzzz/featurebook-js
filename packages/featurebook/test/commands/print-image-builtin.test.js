/* eslint-disable camelcase */
/* eslint-disable quotes */
const path = require('path');
const {
  Iwant, given, when, then_wait_at_most, expect, before, after,
  rootdir, node, cleanupFile,
  content_is_valid_pdf, buffer_length_between,
  one_minute,
} = require('../test.helpers');

const getOutputFile = (outputDir) => {
  const outputPath = outputDir.startsWith('.') ? path.join(rootdir, outputDir) : outputDir;
  return path.join(outputPath, 'specification.pdf');
};

const folder = '../../test/print-image-builtin';
const outputDir = `${folder}/dist/pdf`;
const outputFile = getOutputFile(outputDir);
const cleanup = async () => cleanupFile(outputFile);

given('I have spec with a builtin image', () => {
  when('I run command ‘print’', () => {
    then_wait_at_most(one_minute)
      .that('it generates a valid pdf file', async () => {
        const output = await node({
          program: 'bin/featurebook',
          options: `print -o ${outputDir} ${folder}`,
          outputFile,
        });

        Iwant(output)
          .and.to.satisfy(buffer_length_between(18000, 19000))
          .and.to.satisfy(content_is_valid_pdf);
      });

    before(cleanup);
    after(cleanup);
  });
});

given('I have spec with a builtin image', () => {
  when('I dry-run command ‘print’', () => {
    then_wait_at_most(one_minute).that('it generates expected output to the console', async () => {
      const str = await node({
        program: 'bin/featurebook',
        options: `print --dry-run -o ${outputDir} ${folder}`,
      });

      expect(str).toMatchSnapshot();
    });
  });
});
