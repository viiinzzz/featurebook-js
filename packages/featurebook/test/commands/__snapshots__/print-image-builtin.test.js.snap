// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`I have spec with a builtin image I dry-run command â€˜printâ€™ it generates expected output to the console 1`] = `
"featurebook v3.0.0-b.9 pdf:
[90m..\\..\\test\\print-image-builtin\\dist\\pdf\\specification.pdf[39m
done.
"
`;
